app/services/twilio_inbound_call_manager.rb

```diff
@@ -0,0 +1,115 @@
+require "goodcity/redis"
+
+class TwilioInboundCallManager
+
+  DonorPrefix  = "twilio_donor"
+  NotifyPrefix = "twilio_notify"
+
+  def initialize(options = {})
+    @admin_mobile = options[:admin_mobile]
+    @user_id      = options[:user_id]
+  end
+
+  def self.caller_has_active_offer?(mobile)
+    return false if mobile.blank?
+    return false if (donor = User.find_by_mobile(mobile)).nil?
+    donor_active_offers = donor.offers.non_draft
+    staff_active = Version.past_month_activities(donor_active_offers, donor.id)
+    return (donor_active_offers.count.zero? && staff_active.count.zero?)
+  end
+
+  def set_mobile
+    store(donor_id_key, @admin_mobile)
+  end
+
+  def mobile
+    redis.get(donor_id_key)
+  end
+
+  def call_teardown
+    redis.del(donor_id_key)
+    redis.del(notify_key)
+  end
+
+  def notify_incoming_call
+    # notify only once when at least one worker is offline
+    if redis.get(notify_key).blank?
+      send_calling_notification
+      incoming_call_version
+      store(notify_key, true)
+    end
+  end
+
+  def notify_accepted_call
+    send_call_accept_notification
+    call_accept_version
+  end
+
+  private
+
+  def notify_key
+    "#{NotifyPrefix}_#{@user_id}"
+  end
+
+  def donor_id_key
+    "#{DonorPrefix}_#{@user_id}"
+  end
+
+  def store(key, value)
+    redis.set(key, value)
+    redis.expireat(key, Time.now.to_i + 60)
```

Comment:
@swatijadhav Looks like you can send the expireat key when you set the value. See https://github.com/redis/redis-rb/blob/98e3e7a516fc9b4609bc8ab482605f835a4de621/lib/redis.rb#L657